util/new_protocol_target.py

```diff
@@ -60,45 +60,80 @@ def main():
-    '`manticore::protocol`.'
+    'message.'
   )
-  argparser.add_argument('typename',
-                         type=str,
-                         help='the name of the type to fuzz, such as '\
-                              'firmware_version::FirmwareVersionRequest')
-  argparser.add_argument('--target-templates',
-                         type=str,
-                         choices=['to_wire', 'from_wire', 'to_wire_fuzz_safe'],
-                         nargs='*',
-                         default=['to_wire', 'from_wire'],
-                         help='which target templates to use; defaults to all')
-  argparser.add_argument('--fuzz-dir',
-                         type=str,
-                         default='fuzz',
-                         help='the fuzzing directory, relative to the project'\
-                              'root')
+  subcommands = argparser.add_subparsers(
+    required=True,
+    dest='subcommand',
+    help='subcommands')
+
+  argparser_generate = subcommands.add_parser(
+    'generate',
+    help='generate new fuzz targets')
+  argparser_generate.add_argument(
+    'typename',
+    type=str,
+    help='the name of the type to fuzz, such as '\
+         'firmware_version::FirmwareVersionRequest')
+  argparser_generate.add_argument(
+    '--fuzz-dir',
+    type=str,
+    default='fuzz',
+    help='the fuzzing directory, relative to the project root')
+  argparser_generate.add_argument(
+    '--target-templates',
+    type=str,
+    nargs='*',
+    choices=['to_wire', 'from_wire', 'to_wire_fuzz_safe'],
+    default=['to_wire', 'from_wire'],
+    help='which target templates to use; defaults to all')
+
+  argparser_ci = subcommands.add_parser(
+      'ci',
+      help='generate CI jobs for a set of fuzz targets for one command type')
+  argparser_ci.add_argument(
+      'message_type',
+      type=str,
+      help='prefix for the four fuzz targets to generate a job for, such as'\
+           'firmware_version')
+  argparser_ci.add_argument(
+      '--workflow-file',
+      type=str,
+      default='.github/workflows/fuzz.yml',
+      help='the workflow file, relative to the project root')
+
   args = argparser.parse_args()
 
   util_dir = os.path.dirname(__file__)
-  fuzz_dir = os.path.join(os.path.dirname(util_dir), args.fuzz_dir)
-  (module, typename) = tuple(args.typename.split('::'))
-  typeid = re.search('^\w+', typename)[0]  # Remove a trailing <'static>
-  snake_case_name = camel_to_snake(typeid)
+  repo_top = os.path.dirname(util_dir)
   command = ' '.join(sys.argv)
 
-  renames = {'to_wire_fuzz_safe': 'to_wire'}
-  for template_name in args.target_templates:
-    template = read_file(os.path.join(
-        util_dir, '{}.rs.template'.format(template_name)))
+  if args.subcommand == 'generate':
+    fuzz_dir = os.path.join(repo_top, args.fuzz_dir)
+    (module, typename) = tuple(args.typename.split('::'))
+    typeid = re.search('^\w+', typename)[0]  # Remove a trailing <'static>
```

Comment:
You could use `match` and remove `^` from the pattern instead of using `search` with `^`.